src/dft/backends/mklgpu/commit.cpp

```diff
@@ -24,27 +24,110 @@
 #endif
 
 #include "oneapi/mkl/types.hpp"
+#include "oneapi/mkl/exceptions.hpp"
 
+#include "oneapi/mkl/dft/detail/commit_impl.hpp"
+#include "oneapi/mkl/dft/detail/types_impl.hpp"
+#include "oneapi/mkl/dft/detail/descriptor_impl.hpp"
 #include "oneapi/mkl/dft/detail/mklgpu/onemkl_dft_mklgpu.hpp"
 
+#include "dft/backends/mklgpu/mklgpu_helpers.hpp"
+
+// MKLGPU header
+#include "oneapi/mkl/dfti.hpp"
+
+/**
+Note that in this file, the Intel MKL-GPU library's interface mirrors the interface
+of this OneMKL library. Consequently, the types under dft::TYPE are closed-source MKL types, 
+and types under dft::detail::TYPE are from this library.
+**/
+
 namespace oneapi {
 namespace mkl {
 namespace dft {
 namespace mklgpu {
+namespace detail {
 
-void commit_f(descriptor<precision::SINGLE, domain::REAL> &desc, sycl::queue &queue) {
-    throw std::runtime_error("Not implemented for mklgpu");
-}
-void commit_c(descriptor<precision::SINGLE, domain::COMPLEX> &desc, sycl::queue &queue) {
-    throw std::runtime_error("Not implemented for mklgpu");
-}
-void commit_d(descriptor<precision::DOUBLE, domain::REAL> &desc, sycl::queue &queue) {
-    throw std::runtime_error("Not implemented for mklgpu");
-}
-void commit_z(descriptor<precision::DOUBLE, domain::COMPLEX> &desc, sycl::queue &queue) {
-    throw std::runtime_error("Not implemented for mklgpu");
+/// Commit impl class specialization for MKLGPU.
+template <dft::detail::precision prec, dft::detail::domain dom>
+class commit_derived_impl : public dft::detail::commit_impl {
+private:
+    // Equivalent MKLGPU precision and domain from OneMKL's precision / domain.
+    static constexpr dft::precision mklgpu_prec = to_mklgpu(prec);
+    static constexpr dft::domain mklgpu_dom = to_mklgpu(dom);
+    using mklgpu_descriptor_t = dft::descriptor<mklgpu_prec, mklgpu_dom>;
+
+public:
+    commit_derived_impl(sycl::queue queue, dft::detail::dft_values config_values)
+            : oneapi::mkl::dft::detail::commit_impl(queue),
+              handle(config_values.dimensions) {
+        set_value(handle, config_values);
+        // MKLGPU does not throw an informative exception for the following:
+        if constexpr (prec == dft::detail::precision::DOUBLE) {
+            if (!queue.get_device().has(sycl::aspect::fp64)) {
+                throw mkl::exception("DFT", "commit", "Device does not support double precision.");
+            }
+        }
+
+        try {
+            handle.commit(queue);
+        }
+        catch (const std::exception& mkl_exception) {
+            // Catching the real MKL exception causes headaches with naming.
+            throw mkl::exception("DFT", "commit", mkl_exception.what());
```

Comment:
Commiting a descriptor with multiple dimensions does not seem to be working at the moment. Hitting this exception I believe. But it fails also on the cpu implementation

Reply:
This is an issue in the computation of the default strides, discussed in another PR: https://github.com/hjabird/oneMKL/pull/3